numpyro/infer/util.py

```diff
@@ -795,8 +795,17 @@ def single_prediction(val):
                 **model_kwargs,
             )
         else:
+
+            def _samples_wo_deterministic(msg):
+                return (
+                    samples.get(msg["name"]) if msg["type"] != "deterministic" else None
+                )
+
             model_trace = trace(
-                seed(substitute(masked_model, samples), rng_key)
+                seed(
+                    substitute(masked_model, substitute_fn=_samples_wo_deterministic),
```

Comment:
Could you help me change line `777` to `condition` so that? It would be nice to add an argument like `exclude_deterministic_from_posterior` to `Predictive` to maintain two behaviors. We will pass such argument to this `_predictive` function to control the behavior.

Reply:
I added the conditional logic, I tried the name `exclude_deterministic_params` because it was shorter but let me know if that's insufficient

For this could you elaborate?
> Could you help me change line 777 to condition so that? 

Would I be changing L777 from `substitute` to `condition`?

And should I add the deterministic fix there as well?

Reply:
Yup, using condition there is fine because we don't substitute deterministic sites under the`condition` handler.

Reply:
ok just switched it to `condition`